src/scriptherder

```diff
@@ -738,6 +738,36 @@ def mode_check(args, logger):
     return exit_status['UNKNOWN']
 
 
+def mode_lastlog(args, logger):
+    """
+    View script output for the last execution for either a specific
+    job, or all jobs.
+
+    @param args: Parsed command line arguments
+    @param logger: logging logger
+    """
+    _jobs = sorted(_get_job_results(args, logger), key=lambda x: x.start_time)
+    jobs_by_name = {}
+    for job in _jobs:
+        if job.name not in jobs_by_name:
+            jobs_by_name[job.name] = []
+        jobs_by_name[job.name].append(job)
+
+    if len(jobs_by_name) > 0:
+        if args.cmd and args.cmd[0] in jobs_by_name:
+            last_job = jobs_by_name[args.cmd[0]][-1]
+            if last_job.output_filename:
+                subprocess.call(['less', last_job.output_filename])
+            else:
+                print('No script output found for job {!s}'.format(last_job.name))
+        else:
+            for (name, jobs) in jobs_by_name.items():
+                if jobs[-1].output_filename:
+                    subprocess.call(['less', jobs[-1].output_filename])
```

Comment:
Convenience is good, but I always think really hard before invoking external commands. Maybe the data (output_filename) can be trusted in this case, but maybe not. In my installations all users can create scriptherder job outputs, so this could be a way to trick root into viewing arbitrary files using a pager... hmm...